tensorflow/core/kernels/ctc_loss_op.cc

```diff
@@ -186,4 +220,152 @@ REGISTER_CPU(double);
 
 #undef REGISTER_CPU
 
+#if GOOGLE_CUDA
+class CTCLossOpGPU : public OpKernel {
+
+ public:
+  explicit CTCLossOpGPU(OpKernelConstruction* ctx) : OpKernel(ctx) {
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("preprocess_collapse_repeated",
+                                     &preprocess_collapse_repeated_));
+    OP_REQUIRES_OK(ctx,
+                   ctx->GetAttr("ctc_merge_repeated", &ctc_merge_repeated_));
+    OP_REQUIRES_OK(ctx, ctx->GetAttr("ignore_longer_outputs_than_inputs",
+                                     &ignore_longer_outputs_than_inputs_));
```

Comment:
Since these are always required to be false why not move the 

```
    OP_REQUIRES(ctx, preprocess_collapse_repeated_ == false,
                errors::InvalidArgument("GPU CTCLossOp requires "
                                        "preprocess_collapse_repeated to be "
                                        "false"));
```

etc. checks here and remove the variables in the class?